.github/workflows/ci_tests_run_notebooks.yml

```
@@ -0,0 +1,77 @@
+name: Test notebooks
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+  schedule:
+    - cron: '0 5 * * 1'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  tests:
+    # Do not run the test matrix on PRs affecting the rendering
+    if: ${{ !(github.event_name == 'pull_request') || !contains(github.event.pull_request.labels.*.name, 'html rendering / skip testing') }}
+
+    name: ${{ matrix.os }} ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        # Run all supported OS for one Python version, then add a few extra scenarios
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ['3.12']
+        toxenv: [py312-test]
+        name: ['with Python 3.12',]
+
+        include:
+          - python-version: '3.10'
+            toxenv: py310-test-oldestdeps
+            name: with Python 3.10 and oldest versioned dependencies
+            os: ubuntu-latest
+
+          - python-version: '3.13'
+            toxenv: py313-test-devdeps
+            name: with Python 3.13 and developer versioned dependencies
+            os: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332  # v4.1.7
```

For official GH actions, the version that requires less maintenance is nice.

```suggestion
    - uses: actions/checkout@v4
```